dev_container

```diff
@@ -816,6 +816,143 @@ launch() {
         "$@"
 }
 
+build_and_install_desc() { c_echo 'Build the application and install the binaries into install/<app> if supported
```

Comment:
It looks like this overlaps heavily with `build_and_run`. To reduce the maintenance burden and avoid redundancy, could we please instead add a `run_build_args` option or similar to `build_and_run` to enable passing the `--install` flag? Possibly along with a `--no_run` flag if you'd like to avoid the requirement to launch the app.

Reply:
In the meantime, can we please revert app instructions and QA isntructions to rely on `dev_container build` and `run build` directly for now?

Reply:
Let me find a way to consolidate the two as I think UX is also important here:
`./dev_container build_and_install my_app`
vs
`./dev_container build_and_run my_app --install --no_run` 

:rofl:  I guess I'm too lazy to type those few extra characters :rofl:

Reply:
My concern is that we do not want to start introducing `<action1>_and_<action2>` commands every time we want to chain together multiple behaviors. The `build_and_run` scope was originally intended as a convenience to cover the majority of app build cases, with `dev_container` and `run` as the appropriate fallback for more advanced app cases, such as the case of installation.

I'm not opposed to streamlining additional cases, but we should do it in a way that doesn't require introducing new CLI entrypoints to `dev_container` with each action.

Given that the motivation for this change is time sensitive, my request is that we fall back to `dev_container` and `run` for now, and continue discussion on advanced UX once QA can move forward.

Reply:
Ok, I have updated `build_and_install` to reuse `build_and_run` with a couple of new options. Please have a look.

Reply:
Awesome, having `build_and_install` as a streamlined interface to `build_and_run` looks great and is OK for moving forward. Thanks for refactoring.